framework/include/userobject/SideUserObject.h

```diff
@@ -48,4 +52,47 @@ class SideUserObject : public UserObject,
   const Real & _current_side_volume;
 
   const BoundaryID & _current_boundary_id;
+
+  /**
+   * Returns the local FaceInfo to use in functor arguments and interpolations.
+   * Note this face info could hold the element from the other side of the
+   * sideset. Sidesets are oriented!
+   */
+  const FaceInfo * getFaceInfo()
+  {
+
+    // Either the element or the neighbor is a valid argument to get a face info
+    const Elem * neighbor = _current_elem->neighbor_ptr(_current_side);
+    const Elem * element = _current_elem;
+    int side = _current_side;
+
+    mooseAssert(_current_elem, "We should have an element");
+    mooseAssert(_current_elem->active(), "The current element should be active");
+
+    // No neighbor means we are at a boundary, a FaceInfo exists in the mesh
+    // If a neighbor exists, the face info may only be defined on that side
+    if (neighbor)
+    {
+      // neighbor is not active and we cant just pick an arbitrary refined element in neighbor
+      // no FaceInfo in mesh will satisfy us, let's error out
+      if (!neighbor->active())
+        mooseError("Inactive neighbor found in SideUO loop. Uneven refinement across sidesets "
+                   "is not supported by the faceInfo system in user objects");
+
+      // First check refinement level
+      if (_current_elem->level() < neighbor->level())
+      {
+        element = neighbor;
+        side = neighbor->which_neighbor_am_i(_current_elem);
+      }
+      // Then check ids
+      else if (_current_elem->id() > neighbor->id())
+      {
+        element = neighbor;
+        side = neighbor->which_neighbor_am_i(_current_elem);
+      }
+    }
+
+    return _mesh.faceInfo(element, side);
+  }
```

Comment:
```suggestion
void
getFaceInfo()
{
  _face_infos.clear();

  // Either the element or the neighbor is a valid argument to get a face info
  const Elem * equal_level_neighbor = _current_elem->neighbor_ptr(_current_side);
  _candidate_neighbors = {equal_level_neighbor};

  mooseAssert(_current_elem, "We should have an element");
  mooseAssert(_current_elem->active(), "The current element should be active");

  // No neighbor means we are at a boundary, a FaceInfo exists in the mesh
  // If a neighbor exists, the face info may only be defined on that side
  if (equal_level_neighbor)
  {
    // neighbor is not active and we cant just pick an arbitrary refined element in neighbor
    // no FaceInfo in mesh will satisfy us, let's error out
    if (!equal_level_neighbor->active())
      equal_level_neighbor->active_family_tree_by_neighbor(_candidate_neighbors, _current_elem);

    for (const Elem * neighbor : _candidate_neighbors)
    {
      const Elem * element = _current_elem;
      auto side = _current_side;

      // First check refinement level
      if (_current_elem->level() < neighbor->level())
      {
        element = neighbor;
        side = neighbor->which_neighbor_am_i(_current_elem);
      }
      // Then check ids
      else if (_current_elem->id() > neighbor->id())
      {
        element = neighbor;
        side = neighbor->which_neighbor_am_i(_current_elem);
      }
      _face_infos.push_back(_mesh.faceInfo(element, side));
    }
  }
  else
    _face_infos.push_back(_mesh.faceInfo(_current_elem, _current_side));
}
```

Reply:
Oopsie, I didn't update the comments, but I think the above should do what you need for mixed level refinement. This should also go in the .C file

Reply:
this means that in every postprocessor you will have to nest a loop on the faceInfos. This is a little out of line with a loop on quadrature points imo

Reply:
umm. Maybe I am being too narrow minded. Like the faceInfo loop will only be triggered for functors that work with FaceInfos. 
The functors that work with qp on the side (FE variables) will not need faceInfos

Reply:
That seems like a pretty straightforward fix. Call `getFaceInfo()` in `execute` if `_fe_problem.haveFV()`

Reply:
Oh never mind. I see that looping over face info for a 1-point quadrature rule on a coarse face doesn't make much sense

Reply:
However, I think that you could relax you error somewhat. If the `_current_elem` is more refined than its active neighbor, then you could proceed. But if you don't want to support that case right now, I'll accept that

Reply:
Yea my shower thoughts were that in implementations where you are making functor evaluations based on face-information, we could
1.  keep my suggested code changes above
2. Still call `getFaceInfo` in the derived class (don't worry about modifying `execute` farther down)
3. Return from `computeQpIntegral` early with 0, if `_qp != 0`

But the suggestion of actually having a base class that does loops over face information is much better IMO